### PR TITLE
Refactor SubscribeButton and improve subscription flow [4/n]

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

This PR refactors the `SubscribeButton` component, introduces the `SubscribeModal`, and improves the overall subscription flow. The changes include:

1. Wrapping the `SubscribeButton` component with a `ModalProvider` to handle the subscribe modal.
2. Adding the `SubscribeModal` component to the `SubscribeButton` component.
3. Modifying the `useSubscribeButton` hook to utilize the `ModalContext` and improve the subscription toggle logic.
4. Updating the `WalletSignIn` hook to check if the current session user is the current wallet connected.

The primary goal of these changes is to enhance the subscription experience for users, ensuring that they have a valid email address before subscribing to notifications. The refactoring also simplifies the code structure and makes it easier to maintain.

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] (if relevant) I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] (if relevant) I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3317** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3317" target="_blank">Graphite</a> 
    * **PR #3318** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3318" target="_blank">Graphite</a> 
      * **PR #3319** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3319" target="_blank">Graphite</a> 
        * **PR #3321** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3321" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3321?utm_source=stack-comment).

File: src/components/SubscribeButton/hooks/useSubscribeModal.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+type Values = Yup.InferType<typeof Schema>
+
+export const useSubscribeModal = () => {
+  const modal = useContext(ModalContext)
+  const initialValues: Values = { email: '' }
+
+  const onSubmit = useCallback(
+    async (values: Values, { setSubmitting }: FormikHelpers<Values>) => {
+      try {
+        await axios.post('/api/account/update-details', values)
```
Comment (tomquirk) [inline src/components/SubscribeButton/hooks/useSubscribeModal.ts:29]: maybe should be simply `axios.put(/api/account)`

File: src/hooks/WalletSignIn.ts [hunk 2]
```diff
@@ -9,6 +9,23 @@ export const useWalletSignIn = () => {
...
+      const { error, data } = await supabase
+        .from('users')
+        .select('wallet')
+        .eq('id', session.user.id)
+        .single()
+      if (error) {
+        console.error(error)
+        return false
+      }
+      return data.wallet === wallet.userAddress.toLowerCase()
```
Comment (tomquirk) [inline src/hooks/WalletSignIn.ts:24]: [isEqualAddress](https://github.com/jbx-protocol/juice-interface/blob/584bd7ab27a64733f563cb552868061fe1b11f5a/src/utils/address.ts#L21)

**Review by tomquirk — APPROVED** [review]
lurv it

**Comment by wraeth-eth** [comment]
@wraeth-eth queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3321).

**Comment by wraeth-eth** [comment]
[Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3321) couldn't merge this pull request because a downstack PR #3319 failed to merge.

**Comment by wraeth-eth** [comment]
@wraeth-eth queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3321).

**Comment by wraeth-eth** [comment]
[Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3321) rebased this pull request as part of a merge.

**Comment by wraeth-eth** [comment]
@wraeth-eth merged this pull request with [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3321).

**Pull request merged**